### DISCUSSION TITLE
How to skip the npm-install step in git+ssh dependencies

### DISCUSSION THREAD

**Discussion by JoseGalRe** [pnpm/pnpm#8284] [Q&A]
Hi, i have a private repo that i need to use as a dependency

i run
```
pnpm add git+ssh://git@ssh.dev.azure.com:v3/.....#develop
```

![image](https://github.com/pnpm/pnpm/assets/1406297/9423331f-0d1a-4d7c-bde9-519db4e64cb0)

The problem, when the dependency is being created, it executed `Running npm-install script`, This step fails because the repository uses a subdependency that requires a specific version of node, but in my case, I don't need the sub dependencies to be installed, since I only need some constants from this git+ssh repo.

so, my question.

Is there a way to not run `Running npm-install script` and just leave the git+ssh repo cloned as is?

Thanks in advance.

**Comment by zkochan** [comment]
Try with `--ignore-scripts`.

**Reply by JoseGalRe** [reply]
thanks, works fine, when git+ssh repo is already in package.json and the run `pnpm i` on clean install (without pnpm store or node_modules) the script is executed.

In my project I have dependencies that do need to execute scripts like postinstall, so using `--ignore-scripts` when it is a clean installation is not convenient, is there a way to bypass only one dependency?

![image](https://github.com/pnpm/pnpm/assets/1406297/d348360d-0d77-404c-b8a6-0c1f0d88cf19)

**Reply by zkochan** [reply]
Try this: https://pnpm.io/package_json#pnpmneverbuiltdependencies

I don't know if it works with git-hosted deps.

**Reply by JoseGalRe** [reply]
> Try this: https://pnpm.io/package_json#pnpmneverbuiltdependencies
> 
> I don't know if it works with git-hosted deps.

thanks, i tried with

```
{
  "dependencies": {
    "package-name": "git+ssh://git@ssh.dev.azure.com:v3/...#develop"
  },
  "pnpm": {
    "neverBuiltDependencies": [
      "package-name",
      "package-name@git+ssh://git@ssh.dev.azure.com:v3/...#develop"
    ]
  }
}
```

but it doesn't work, should by nice if the git+ssh dependency could use the `--ignore-scripts` in params

like

```
{
  "dependencies": {
    "package-name": "git+ssh://git@ssh.dev.azure.com:v3/...#develop&ignore-scripts"
  },
}
```
